app/rtc_compare.py

```diff
@@ -0,0 +1,386 @@
+    '''
+    Compare the dataset or attribute defined by `str_key`
+    NOTE: For attributes, the path and the key are separated by newline character ('\n')
+
+    Parameters:
+    -----------
+    hdf5_obj_1: h5py.Group
+        The 1st HDF5 object to compare
+    hdf5_obj_2: h5py.Group
+        The 2nd HDF5 object to compare
+    str_key: str
+        Key to the dataset or attribute
+    is_attr: bool
+        Designate if `str_key` is for dataset or attribute
+
+    Return:
+    -------
+    _: True when the dataset are identical; False otherwise
+    '''
+
+    # Prepare to comapre the data in the HDF objects
+    if is_attr:
+        # str_key is for attribute
+        path_attr, key_attr = str_key.split('\n')
+        val_1 = hdf5_obj_1[path_attr].attrs[key_attr]
+        val_2 = hdf5_obj_2[path_attr].attrs[key_attr]
+        # Force the datatype of val_1 and _2 to make use of numpy features
+        if not isinstance(val_1,np.ndarray):
+            val_1 = np.array(val_1)
+
+        if not isinstance(val_2,np.ndarray):
+            val_2 = np.array(val_2)
+
+    else:
+        # str_key is for dataset
+        val_1 = np.array(hdf5_obj_1[str_key])
+        val_2 = np.array(hdf5_obj_2[str_key])
+
+    # convert oeject reference to the path to which it is pointing
+    if (len(val_1.shape)>=1) and ('shape' in dir(val_1[0])):
+        if isinstance(val_1[0], np.void) or\
+        ((len(val_1[0].shape) == 1) and (isinstance(val_1[0][0], h5py.h5r.Reference))):
+            # Example:
+            # attribute `REFERENCE_LIST` in
+            # /science/CSAR/RTC/grids/frequencyA/xCoordinates'
+            # attribute `DIMENSION_LIST` in
+            # /science/CSAR/RTC/grids/frequencyA/VH
+            list_val_1 = list(itertools.chain.from_iterable(val_1))
+            val_1_new = [None] * len(list_val_1)
+            for i_val, element_1 in enumerate(list_val_1):
+                if isinstance(element_1, h5py.h5r.Reference):
+                    print('    - Object reference found.')
+                    val_1_new[i_val] = np.str_(hdf5_obj_1[element_1].name)
+                else:
+                    val_1_new[i_val] = element_1
+            val_1 = np.array(val_1_new)
+
+    #Repeat the same thing for val_2
+    if (len(val_2.shape)>=1) and ('shape' in dir(val_2[0])):
+        if isinstance(val_2[0], np.void) or\
+        ((len(val_2[0].shape) == 1) and (isinstance(val_2[0][0], h5py.h5r.Reference))):
+            list_val_2 = list(itertools.chain.from_iterable(val_2))
+            val_2_new = [None] * len(list_val_2)
+            for i_val, element_2 in enumerate(list_val_2):
+                if isinstance(element_2, h5py.h5r.Reference):
+                    print('    - Object reference found.')
+                    val_2_new[i_val] = np.str_(hdf5_obj_2[element_2].name)
+                else:
+                    val_2_new[i_val] = element_2
+            val_2 = np.array(val_2_new)
```

Comment:
Moved the reference -> path string conversion right after loading the data from the HDF5 object.